packages/go_router/example/lib/shell_route_top_route.dart

```diff
@@ -0,0 +1,326 @@
+                name: 'b.details',
+                path: 'details',
+                builder: (BuildContext context, GoRouterState state) {
+                  return const DetailsScreen(label: 'B');
+                },
+              ),
+            ],
+          ),
+
+          /// The third screen to display in the bottom navigation bar.
+          GoRoute(
+            // The name of this route used to determine the title in the ShellRoute.
+            name: 'c',
+            path: '/c',
+            builder: (BuildContext context, GoRouterState state) {
+              return const ScreenC();
+            },
+            routes: <RouteBase>[
+              // The details screen to display stacked on the inner Navigator.
+              // This will cover screen C but not the application shell.
+              GoRoute(
+                // The name of this route used to determine the title in the ShellRoute.
+                name: 'c.details',
+                path: 'details',
+                builder: (BuildContext context, GoRouterState state) {
+                  return const DetailsScreen(label: 'C');
+                },
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      routerConfig: _router,
+    );
+  }
+}
+
+/// Builds the "shell" for the app by building a Scaffold with a
+/// BottomNavigationBar, where [child] is placed in the body of the Scaffold.
+class ScaffoldWithNavBar extends StatelessWidget {
+  /// Constructs an [ScaffoldWithNavBar].
+  const ScaffoldWithNavBar({
+    required this.title,
+    this.routeName,
+    required this.child,
+    super.key,
+  });
+
+  /// The title to display in the AppBar.
+  final String title;
+
+  /// The name of the current route.
+  final String? routeName;
+
+  /// The widget to display in the body of the Scaffold.
+  /// In this sample, it is a Navigator.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    final PreferredSizeWidget leadingButton = _buildAppBar(context);
+    return Scaffold(
+      body: child,
+      appBar: leadingButton,
+      bottomNavigationBar: BottomNavigationBar(
+        items: const <BottomNavigationBarItem>[
+          BottomNavigationBarItem(
+            icon: Icon(Icons.home),
+            label: 'A Screen',
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.business),
+            label: 'B Screen',
+          ),
+          BottomNavigationBarItem(
+            icon: Icon(Icons.notification_important_rounded),
+            label: 'C Screen',
+          ),
+        ],
+        currentIndex: _calculateSelectedIndex(context),
+        onTap: (int idx) => _onItemTapped(idx, context),
+      ),
+    );
+  }
+
+  /// Animates the AppBar title and leading button when it changes.
```

Comment:
We might not need to have an animated app bar for this sample, if we can simplify the code to just `AppBar(title: _getScreenTitle(context),)` that might be easier from a readability point of view.

Reply:
That makes sense, I've removed the animated related code, though it still needs the custom leading back button